src/OrchardCore.Modules/OrchardCore.Media/Shortcodes/AssetUrlShortcodeProvider.cs

```diff
@@ -55,7 +55,16 @@ public ValueTask<string> EvaluateAsync(string identifier, Arguments arguments, s
                 }
                 else
                 {
-                    content = _mediaFileStore.MapPathToPublicUrl(content);
+                    var queryIndex = content.IndexOf('?');
+                    string queryString = null;
+
+                    if (queryIndex >= 0)
+                    {
+                        queryString = content[queryIndex..];
+                        content = content[..queryIndex];
+                    }
+
+                    content = _mediaFileStore.MapPathToPublicUrl(content) + queryString;
```

Comment:
Can we create a helper for than instead of duplicate the code

Reply:
I'm not particularly sold on this change. I've done it to continue allowing to use something like `[asset_url]foo.jpeg?width=100[/asset_url]`, which was possible before. But the correct usage would be to use an argument, e.g. `[asset_url width="100"]foo.jpeg[/asset_url]`.

Maybe we should just treat any content as a complete media path, i.e. escaping any query parameter here as well. What do you think?

Reply:
I create a new helper in `StringUriExteions` and I decided to keep the functionality for now.